lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -649,4 +678,143 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        }
+
+        // check if object is permanently deleted or marked as deleted
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().
+                                         withIds(Arrays.asList((long)wsid)).
+                                         withMinObjectID(objid-1).
+                                         withMaxObjectID(objid+1));
+        try {
+            boolean objidExists = false;
+
+            final List<List> objList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            for (List obj : objList) {
+                final long id = (long) ((Integer) obj.get(0)).intValue();
+                if (id == objid) {
+                    objidExists = true;
+                    break;
+                }
+            }
+
+            if (!objidExists) {
+                return StatusEvent.
+                        getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                StatusEventType.DELETE_ALL_VERSIONS).
+                        withNullableAccessGroupID(wsid).
+                        withNullableObjectID(Long.toString(objid)).build();
+            }
+        } catch (IOException ex) {
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        } catch (JsonClientException ex) {
+            // really think it should be IndexingException here
+            // but IndexingException has a protected constructor
+            throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage());
+        }
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
+        ev.getAccessGroupId().get().toString() + "/" +
+                        ev.getAccessGroupObjectId().get());
+
+        final List<ObjectSpecification> getInfoInput = new ArrayList<>();
+        getInfoInput.add(os);
+
+        command = new HashMap<>();
+        command.put("command", "getObjectInfo");
+        command.put("params", new GetObjectInfo3Params().withObjects(getInfoInput));
+
+        final GetObjectInfo3Results objInfo;
+        try {
+            objInfo = ws.getClient().administer(new UObject(command))
+                                        .asClassInstance(GetObjectInfo3Results.class);
+            String name = objInfo.getInfos().get(0).getE2();
+            if (!Utils.isNullOrEmpty(name)) {
+                    latestName = name;
+            }
+        } catch (IOException e) {
+            throw handleException(e);
+        } catch (JsonClientException e) {
+            throw handleException(e);
+        }
+
+        // check if permissions have changed and update state of isPublic
+        try {
+            command = new HashMap<>();
```

Comment:
Isn't there a get ws info method that could be used here?

Reply:
Reused.